src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/ReleaseDateService.kt

```diff
@@ -81,33 +81,32 @@ class ReleaseDateService(
 
   fun getHardStopDate(sentenceDateHolder: SentenceDateHolder): LocalDate? {
     val actualReleaseDate = sentenceDateHolder.actualReleaseDate
-    val conditionalReleaseDate = sentenceDateHolder.conditionalReleaseDate ?: return null
+    val conditionalReleaseDate = sentenceDateHolder.conditionalReleaseDate
 
-    val date = chooseDateForHardstop(actualReleaseDate, conditionalReleaseDate)
+    return when {
+      conditionalReleaseDate == null -> null
+      actualReleaseDate != null && isExcludedFromHardstop(actualReleaseDate, conditionalReleaseDate) -> null
+      else -> calculateHardStopDate(conditionalReleaseDate)
+    }
+  }
 
+  private fun calculateHardStopDate(conditionalReleaseDate: LocalDate): LocalDate {
+    val date = if (conditionalReleaseDate.isNonWorkingDay()) 1.workingDaysBefore(conditionalReleaseDate) else conditionalReleaseDate
     return 2.workingDaysBefore(date)
   }
 
+  private fun isExcludedFromHardstop(actualReleaseDate: LocalDate, conditionalReleaseDate: LocalDate): Boolean {
```

Comment:
Is this the replacement for this??
**val isNotAnEarlyRelease = actualReleaseDate >= 1.workingDaysBefore(conditionalReleaseDate)
    val isArdTheReleaseDate = actualReleaseDate <= conditionalReleaseDate**

Reply:
yes kind of - previously we would have used ARD as the basis of hard stop if these were true.

Now we will not have a hard stop date if these are true, we also now don't do hard stop if ARD is > than CRD